systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaNodePoolUtils.java

```diff
@@ -37,4 +45,51 @@ public static void scaleKafkaNodePool(String namespaceName, String kafkaNodePool
         LOGGER.info("Scaling KafkaNodePool: {}/{} to {} replicas", namespaceName, kafkaNodePoolName, scaleToReplicas);
         KafkaNodePoolResource.kafkaNodePoolClient().inNamespace(namespaceName).withName(kafkaNodePoolName).scale(scaleToReplicas);
     }
+
+    public static void waitForKafkaNodePoolDeletion(String namespaceName, String kafkaClusterName, String kafkaNodePoolName) {
+        LOGGER.info("Waiting for deletion of KafkaNodePool: {}/{}", namespaceName, kafkaNodePoolName);
+        TestUtils.waitFor("deletion of KafkaNodePool: " + namespaceName + "/" + kafkaNodePoolName, TestConstants.POLL_INTERVAL_FOR_RESOURCE_READINESS, DELETION_TIMEOUT,
+            () -> {
+                if (KafkaNodePoolResource.kafkaNodePoolClient().inNamespace(namespaceName).withName(kafkaNodePoolName).get() == null &&
+                    StrimziPodSetResource.strimziPodSetClient().inNamespace(namespaceName).withName(kafkaClusterName + "-" + kafkaNodePoolName).get() == null
+                ) {
+                    return true;
+                } else {
+                    cmdKubeClient(namespaceName).deleteByName(KafkaNodePool.RESOURCE_KIND, kafkaNodePoolName);
+                    return false;
+                }
+            },
+            () -> LOGGER.info(KafkaNodePoolResource.kafkaNodePoolClient().inNamespace(namespaceName).withName(kafkaNodePoolName).get()));
+    }
+
+    public static void waitForKafkaNodePoolPodsReady(TestStorage testStorage, String kafkaNodePoolName) {
+        waitForKafkaNodePoolPodsReady(
+            testStorage.getNamespaceName(),
+            testStorage.getClusterName(),
+            kafkaNodePoolName,
+            getKafkaNodePoolReplicaCount(testStorage.getNamespaceName(), kafkaNodePoolName));
+    }
+
+    public static void waitForKafkaNodePoolPodsReady(String namespaceName, String kafkaClusterName, String kafkaNodePoolName) {
+        waitForKafkaNodePoolPodsReady(
+            namespaceName,
+            kafkaClusterName,
+            kafkaNodePoolName,
+            getKafkaNodePoolReplicaCount(namespaceName, kafkaNodePoolName));
+    }
```

Comment:
It's okay to have these methods, but I guess that you should know how many replicas each of the nodepools should have -> so instead of doing another API calls, I would just pass there the number of replicas as a method param (you already have it there, I just think that you should define the number, otherwise you will create some false positives)